manifests/database.pp

```diff
@@ -129,20 +132,27 @@
   $database_host_ip                = $zabbix::params::server_database_host_ip,
   $database_charset                = $zabbix::params::server_database_charset,
   $database_collate                = $zabbix::params::server_database_collate,
+  $database_tablespace             = $zabbix::params::server_database_tablespace,
 ) inherits zabbix::params {
 
   # So lets create the databases and load all files. This can only be
   # happen when manage_database is set to true (Default).
   if $manage_database == true {
+    # Complain if database_tablespace is set and the database_type is not postgresql
+    if ($database_tablespace and $database_type != 'postgresql') {
+      notify { "database_tablespace is set to '${database_tablespace}'. This setting is only useful for PostgreSQL databases.": }
```

Comment:
I'd probably just go with a `fail` here instead of a notify resource.

Reply:
IMHO a stale configuration option is not enough reason to have the catalog compilation fail, but `fail` is fine with me. Your call...

Reply:
As notify isn't idempotent, I personally only tend to use it for cases when something will be fixed in the next puppet run.  eg maybe some puppet needs a fact before it can be part of your catalog, but that fact won't be present until after the first run.

```puppet
include java

if $facts['jdk_version'] {
  # Install something that needs java and needs to know what version.
  # ...
} else {
  notify {'jdk_version fact was missing. Foo was not installed.  Please run puppet agent to complete the configuration.':}
}
```

In this case, I think I'd prefer a hard fail rather than a notify resource reporting a 'change' on each run.  (But I don't feel that strongly either)

Reply:
I asked my colleagues about this and they would use a fail as well, so a fail it will be.